llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp

```diff
@@ -901,6 +901,20 @@ bool AMDGPUTargetLowering::aggressivelyPreferBuildVectorSources(EVT VecVT) const
   return true;
 }
 
+void AMDGPUTargetLowering::CollectTargetIntrinsicOperands(
+    const CallInst &I, SmallVectorImpl<SDValue> &Ops, SelectionDAG &DAG,
+    function_ref<SDValue(const Value *)> getValue) const {
+  if (auto Bundle = I.getOperandBundle(LLVMContext::OB_convergencectrl)) {
```

Comment:
There's nothing AMDGPU specific here. This is general infrastructure so I would expect the target callback to be unnecessary

Reply:
The new SDNode being returned has the AMD-specific GLUE opcode. It's not clear to me that we can decide for all targets whether the LLVM convergence operand bundle should be translated to a glued operand. So this allows each target to return whatever node fits their scheme.